synapse/storage/data_stores/main/admin.py

```diff
@@ -0,0 +1,142 @@
+# -*- coding: utf-8 -*-
+# Copyright 2019 Matrix.org Foundation C.I.C
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import uuid
+from collections import defaultdict
+
+import attr
+from constantly import NamedConstant, Names
+
+from synapse.storage._base import SQLBaseStore
+
+
+class TokenState(Names):
+    VALID = NamedConstant()
+    EXPIRED = NamedConstant()
+    NOT_YET_ACTIVE = NamedConstant()
+    NON_EXISTANT = NamedConstant()
+
+
+@attr.s
+class TokenLookupResult(object):
```

Comment:
some docs on this would be useful. What do the fields mean, and in particular what is the shape of `permissions` and what are example keys for the dicts?